src/item.h

```diff
@@ -1194,25 +1194,47 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
         item * gunmod_find( const itype_id& mod );
         const item * gunmod_find( const itype_id& mod ) const;
 
-        /**
-         * Returns currently active auxiliary (@ref is_auxiliary_gunmod) gun mod item.
-         * May return null if there is no such gun mod or if the gun is not in the
-         * auxiliary mode (@ref is_in_auxiliary_mode).
-         */
-        item * gunmod_current();
-        item const * gunmod_current() const;
-
         /*
          * Checks if mod can be applied to this item considering any current state (jammed, loaded etc.)
          * @param alert whether to display message describing reason for any incompatibility
          * @param effects whether temporary efects (jammed, loaded etc) are considered when checking
          */
         bool gunmod_compatible( const item& mod, bool alert = true, bool effects = true ) const;
 
-        /**
-         * Burst size (see ranged.cpp), includes effects from installed gunmods.
-         */
-        int burst_size() const;
+        struct gun_mode {
+            std::string mode; /** name of this mode */
+            item *target;     /** pointer to base gun or attached gunmod */
+            int qty;          /** burst size or melee range */
+            bool melee;       /** if true perform a melee attach as opposed to shooting */
+
+            operator bool() const { return target != nullptr; }
```

Comment:
> has_ implies a check for existence which any function that fetches an available mode will need to perform and also suggests that the item isn't mutated in doing so.

`has_` implies only a check, not value being returned.

> get_ implies the current mode is returned (which it isn't) or that the object is in someway mutated. It also doesn't have the same implicit suggestion that a mode might not exist.

But it does:
* `vehicle::get_passenger` returns 0 for most parts
* `Character::get_item_position` returns `INT_MIN` for items that aren't there
* `map::get_furn` returns null furniture for furniture-less tiles
* `iuse::get_actor_ptr` returns (returned? not sure if it still happens) null on actor-less iuses
* Items with no quality don't debugmsg on `get_quality`
* `Character::get_ammo` can return empty vector

And so on